apps/workers/bigquery/worker.go

```diff
@@ -102,6 +103,17 @@ func (w *worker) DoTask(ctx context.Context) (task.Result, string) {
 	return task.Failf("unsupported file format %v, expected:sql|json|csv", filepath.Ext(w.File))
 }
 
+// templateQuery lets users template project, dataset and tables values in the query.
+// These value are signified with surrounding brackets
+// {project} {dataset} {table} {src_table} for full path
+func (d Destination) templateQuery(q string) string {
+	q = strings.ReplaceAll(q, "{src_table}", d.String())
+	q = strings.ReplaceAll(q, "{table}", d.Table)
+	q = strings.ReplaceAll(q, "{dataset}", d.Dataset)
+	q = strings.ReplaceAll(q, "{project}", d.Project)
+	return q
+}
+
```

Comment:
There are a couple of things I don't like with this solution. 

1. Its not very flexible. This only allows the user to have one custom template value for project.dataset.table, If more complicated joins are required this solution won't work. 
2. The Destination requires all 3 values to be populated, This makes it annoying if all you need is the dataset. 
3. These values should be cleaned to avoid injection attempts.